doc/guides/coding_guidelines/index.rst

```diff
@@ -0,0 +1,777 @@
+.. _coding_guidelines_misra:
+
+Coding Guidelines
+#################
+
+The project TSC and the Safety Committee of the project agreed to implement
+a staged and incremental approach for complying with a set of coding rules (AKA
+Coding Guidelines) to improve quality and consistency of the code base. Below
+are the agreed upon stages and the approximate timelines:
+
+Stage I
+  Coding guideline rules are available to be followed and referenced,
+  but not enforced. Rules are not yet enforced in CI and pull-requests cannot be
+  blocked by reviewers/approvers due to violations.
+
+Stage II
+  Begin enforcement on a limited scope of the code base. Initially this would be
+  the safety certification scope. For rules easily applied across codebase, we
+  should not limit compliance to initial scope. This step requires tooling and
+  CI setup.
+  This stage will begin during the 2.4 development cycle and end with the Zephyr
+  LTS2 (2.6) to achieve and LTS2 that is ready for certification.
+
+Stage III
+  Revisit the coding guideline and based on experience from previous stages,
+  refine/iterate on selected rules. This stage is to start after LTS2.
+
+Stage IV
+   Expand enforcement to the wider codebase. Exceptions may be granted on some
+   areas of the codebase with a proper justification. Exception would require
+   TSC approval.
+
+.. note::
+
+    Coding guideline rules may be removed/changed at any time by filing a
+    GH issue/RFC.
+
+The coding guideline is based on MISRA-C 2012 and is a subset of MISRA-C. The
+subset is listed in the table below with a summary of the rules, its severity
+and the equivlent rules from other standards for reference.
```

Comment:
we had discussed including a note that funding is available through the project to community members to purchase copies of the misra standard. not sure if we want to state that here or in our upcoming safety project documentation where we point to the codeguideline.

Reply:
need the appropriate wording and what we should say here exactly...

Reply:
@kestewart ...? :)

Reply:
How about something like: 

For existing Zephyr maintainers and collaborators, if you are unable to obtain a copy through your employer, a limited number of copies will be made available through the project.   If you need a copy of MISRA-C 2012, please send email to safety@lists.zephyrproject.org and provide details on reason why you can't obtain one through other options and expected contributions once you have one.   The safety committee will review all requests.

Reply:
thanks @kestewart lgtm